### ISSUE TITLE
Does not compile under Debian 11.5

### ISSUE THREAD

**Issue by marcelr** [protobuf-c/protobuf-c#530]
Hi,

the current git repos does not compile under Debian 11.5   
```
uname 
  5.10.0-19-amd64
g++ --version
  g++ (Debian 10.2.1-6) 10.2.1
```

```
git clone https://github.com/protobuf-c/protobuf-c.git
cd protobuf-c
./autogen.sh
./configure
    protobuf-c 1.4.1

        CC:                     gcc
        CFLAGS:                 -g -O2
        CXX:                    g++ -std=c++11
        CXXFLAGS:               -g -O2
        LDFLAGS:                
        LIBS:                   

        prefix:                 /usr/local
        sysconfdir:             ${prefix}/etc
        libdir:                 ${exec_prefix}/lib
        includedir:             ${prefix}/include
        pkgconfigdir:           ${libdir}/pkgconfig

        bigendian:              no
        protobuf version:       libprotoc 21.9
        
make
  GEN      protobuf-c/protobuf-c.pb.cc
  CXX      protoc-c/protoc_gen_c-c_bytes_field.o
In file included from ./protoc-c/c_field.h:67,
                 from ./protoc-c/c_bytes_field.h:68,
                 from protoc-c/c_bytes_field.cc:63:
/usr/local/include/google/protobuf/stubs/common.h:63:2: error: #error "C++ versions less than C++14 are not supported."
   63 | #error "C++ versions less than C++14 are not supported."
      |  ^~~~~
In file included from ./protoc-c/c_bytes_field.h:68,
                 from protoc-c/c_bytes_field.cc:63:
./protoc-c/c_field.h:108:3: error: ISO C++ forbids declaration of ‘GOOGLE_DISALLOW_EVIL_CONSTRUCTORS’ with no type [-fpermissive]
  108 |   GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(FieldGenerator);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
....
```

Editing Makefile and change to 

```
CXX = g++ -std=c++14
CXXCPP = g++ -E -std=c++14
```
does not help:

```
make
  CXX      protoc-c/protoc_gen_c-c_bytes_field.o
In file included from ./protoc-c/c_bytes_field.h:68,
                 from protoc-c/c_bytes_field.cc:63:
./protoc-c/c_field.h:108:3: error: ISO C++ forbids declaration of ‘GOOGLE_DISALLOW_EVIL_CONSTRUCTORS’ with no type [-fpermissive]
  108 |   GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(FieldGenerator);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./protoc-c/c_field.h:108:37: error: cannot declare parameter to be of abstract type ‘google::protobuf::compiler::c::FieldGenerator’
  108 |   GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(FieldGenerator);
....
```

How to resolve?

Thanks
Marcel

**Comment by MichaelVoelkel** [comment]
Same issue on macOS13.

It seems like an incompatibility between protobuf-c and protobuf? Maybe it would make sense to have protobuf-c versions hard-linked to specific protobuf versions?

**Comment by avrono** [comment]
Same issue here `5.10.0-20-amd64 #1 SMP Debian 5.10.158-2 (2022-12-13) x86_64 GNU/Linux`

**Comment by idoop** [comment]
@avrono @marcelr @MichaelVoelkel  
The reason for these error messages is that protobuf-c did not adapt to the new version of Protobuf in time. we can use older versions of `v3.x.x` to bypass these errors. Like `v3.20.3`:

``` shell
# Debian Bullseye

apt install build-essential -y

# Protobuf
git clone -b v3.20.3 --recurse-submodules https://github.com/protocolbuffers/protobuf.git && cd protobuf 
cmake -B build -S cmake -DCMAKE_BUILD_TYPE=Release -Dprotobuf_BUILD_TESTS=OFF
cmake --build build --config Release -j $(nproc)
cd build && make install

# protobuf-c
git clone https://github.com/protobuf-c/protobuf-c.git  && cd protobuf-c
cmake -B build -S build-cmake -DCMAKE_BUILD_TYPE=Release
cmake --build build --config Release -j $(nproc)
```

**Comment by MichaelVoelkel** [comment]
Yeah, that workaround worked.
But as said, probably the compatibility should be made sure by documenting (or even checking automatically and then warn) which version of protobuf this protobuf-c library is created against on a specific branch/release? Otherwise, every time protobuf itself changes, protobuf-c is potentially broken and manual effort is needed to downgrade protobuf until it finally works

**Comment by JPHutchins** [comment]
> @avrono @marcelr @MichaelVoelkel The reason for these error messages is that protobuf-c did not adapt to the new version of Protobuf in time. we can use older versions of `v3.x.x` to bypass these errors. Like `v3.20.3`:
> 
> ```shell
> # Debian Bullseye
> 
> apt install build-essential -y
> 
> # Protobuf
> git clone -b v3.20.3 --recurse-submodules https://github.com/protocolbuffers/protobuf.git && cd protobuf 
> cmake -B build -S cmake -DCMAKE_BUILD_TYPE=Release -Dprotobuf_BUILD_TESTS=OFF
> cmake --build build --config Release -j $(nproc)
> cd build && make install
> 
> # protobuf-c
> git clone https://github.com/protobuf-c/protobuf-c.git  && cd protobuf-c
> cmake -B build -S build-cmake -DCMAKE_BUILD_TYPE=Release
> cmake --build build --config Release -j $(nproc)
> ```

For completeness, `cd build && make install` after the last step building protobuf-c, above.

**Comment by edmonds** [comment]
Hi,

Please try out the branch in https://github.com/protobuf-c/protobuf-c/pull/673.

Thanks!

**Comment by edmonds** [comment]
This is probably fixed in 1.5.0, please reopen if that's not the case.
